packages/ToggleSwitch/ToggleSwitch.jsx

```diff
@@ -57,48 +66,69 @@ class ToggleSwitch extends Component {
   }
 
   render() {
-    const { id, label, name, value, checked, onBlur, onChange, onFocus, ...rest } = this.props
-
+    const {
+      id,
+      label,
+      name,
+      value,
+      toolTipText,
+      checked,
+      onBlur,
+      onChange,
+      onFocus,
+      toolTipCopy,
+      isLoading,
+      ...rest
+    } = this.props
     const labelledById = `${id}-label`
     const disabled = !!rest && !!rest.disabled
 
-    const switchStatusClass = this.state.checked ? styles.switchOn : styles.switchOff
-    const switchClasses = disabled
-      ? [styles.switch, switchStatusClass, styles.switchDisabled]
-      : [styles.switch, switchStatusClass]
-
+    /* eslint-disable jsx-a11y/label-has-for */
     return (
       <label htmlFor={id}>
-        <Box tag="span" inline between={3}>
-          <span>
-            <input
-              {...safeRest(rest)}
-              className={styles.hiddenInput}
-              id={id}
-              type="checkbox"
-              name={name}
-              value={value}
-              checked={this.state.checked}
-              disabled={disabled}
-              aria-labelledby={labelledById}
-              onChange={this.onChange}
-              onFocus={this.onFocus}
-              onBlur={this.onBlur}
-            />
-
-            <span
-              data-testid={`${id}-switch`}
-              aria-checked={this.state.checked}
-              className={switchClasses.join(' ')}
-            >
-              <span className={styles.slider} />
-            </span>
-          </span>
-
-          <Text id={labelledById} size="medium">
-            {label}
-          </Text>
-        </Box>
+        <FlexGrid>
+          <FlexGrid.Row>
+            <FlexGrid.Col xs={7}>
+              <Box inline between={2}>
+                <Text id={labelledById} size="medium">
+                  {label}
+                </Text>
+                {toolTipText && <Tooltip copy={toolTipCopy}>{toolTipText}</Tooltip>}
+              </Box>
+            </FlexGrid.Col>
+            <FlexGrid.Col xs={1} xsOffset={2} mdOffset={3} lgOffset={3} xlOffset={4}>
+              <InputSwitchWrapper>
+                <HiddenInput
```

Comment:
I know this may seem bit odd at first but we should *NOT* be using a `<Input type="checkbox" />`. That was one of the major consideration of this update. Rather we should be using `<Button />`. Also I know it seems odd to use a `<Button />` as a toggleSwitch, however let's revisit our [design intent](https://github.com/telus/tds-community/issues/112#issuecomment-500551118)

> Toggles should provide immediate results. They are best used for changing the state of system functionalities and preferences.

A checkbox *NEVER* provides immediate results as it's associated to be part of a  `<Form />` & a `<Form />` has a `<button />` or `<input type="submit" />` that submits values, so it's never immediate. This is better explained by _Hayden Pickering_ in his awesome *Inclusive Components* work [seems here](https://inclusive-components.design/toggle-button/) under *THIS DOESN’T QUITE FEEL RIGHT* section.
>The trouble with using form elements is their longstanding association with the collection of data. That is, checkboxes and radio buttons are established as controls for designating values. When a user checks a checkbox, they may just be switching a state, but they may suspect they are also choosing a value for submission.

Also @nicmak really appreciate your contribution(s); this is very much my fault as I was tied with payments project, but I had outlined the technical implementation details [here](https://github.com/telus/tds-community/issues/112#issuecomment-486221017) which I failed to convey before you picked this work up.

Reply:
Changes made, please review , Thanks!